presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -612,6 +613,9 @@ public PlanWithProperties visitExchange(ExchangeNode node, StreamPreferredProper
                         any().withOrderSensitivity(),
                         any().withOrderSensitivity());
             }
+            if (isEnforceFixedDistributionForOutputOperator(session)) {
+                return planAndEnforceChildren(node, fixedParallelism(), fixedParallelism());
```

Comment:
should it be `singleStream()` or `fixedParallelism()` for the purpose of buffering?

Reply:
Fixed stream is better to better utilize parallelism